pgmax/fg/graph.py

```diff
@@ -31,10 +30,19 @@
 from jax.scipy.special import logsumexp
 
 from pgmax.bp import infer
-from pgmax.factors import FAC_TO_VAR_UPDATES
+from pgmax.factors import FAC_TO_VAR_UPDATES, enumeration, logical
 from pgmax.fg import groups, nodes
 from pgmax.utils import cached_property
 
+GROUPS_TO_TYPES: OrderedDict[Type, Type] = collections.OrderedDict(
```

Comment:
Might be better to make this an attribute of the corresponding factor group classes instead of having to maintain a centralized registry, something like
```
factor_type: type = field(init=False, default=enumeration.EnumerationFactor)
```